app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,84 @@
+# Controller to control the flow regarding Articles
+
+class ArticlesController < ApplicationController
+
+  # display articles of all users
+  def index
+    @article = Article.all
+  end
+
+  # User Login Page showing his articles and blog 
+  def main  
+    @user = User.find(params[:user_id])
```

Comment:
use memoization and define user and article into separate functions itself to save db calls